### PR TITLE
mo/devise

### PR DESCRIPTION
**Pull request by kos1kov**

### CODE REVIEW

**Comment by mesmerze** [comment]
Please, do not mess around so many changes from different users with a different meaning in one PR...

[huuuuuge](https://www.youtube.com/watch?v=ddJ8YYl7q-A)

**Comment by mesmerze** [comment]
At first, I need to understand the purpose of this PR. Please edit PR title and description.

File: .hound.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+fail_on_violations: true
+
+rubocop:
+  config_file: .rubocop.yml
+
+reek:
+  enabled: true
+
```
Comment (mesmerze) [inline .hound.yml]: Please remove this line

File: Gemfile [hunk 2]
```diff
@@ -43,6 +45,9 @@ gem 'bootsnap', '>= 1.1.0', require: false
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
```
Comment (mesmerze) [inline Gemfile]: You already have `pry`. Why we need this debugger?

File: config/application.rb [hunk 3]
```diff
@@ -9,11 +9,21 @@
...
     # Initialize configuration defaults for originally generated Rails version.
-    config.load_defaults 5.2
+    # config.load_defaults 5.2
 
     # Settings in config/environments/* take precedence over those specified here.
     # Application configuration can go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded after loading
     # the framework and any gems in your application.
+    # config.api_only = true
+      config.middleware.insert_before 0, Rack::Cors do
```
Comment (mesmerze) [inline config/application.rb]: wrong indentation

File: config/application.rb [hunk 4]
```diff
@@ -9,11 +9,21 @@
...
+        allow do
+          origins 'http://localhost:3000'
+          resource '*', 
+           :headers => :any, 
+           :expose => ['access-token', 'expiry', 'token-type', 'uid', 'client'],
+           :methods => [:get, :post, :patch, :delete, :options]
```
Comment (mesmerze) [inline config/application.rb]: %i[get post patch delete options]

options???

File: db/migrate/20190805091526_devise_token_auth_create_users.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      t.json :tokens
+
+      t.timestamps
+    end
+
+    add_index :users, :email,                unique: true
+    add_index :users, [:uid, :provider],     unique: true
+    add_index :users, :reset_password_token, unique: true
+    add_index :users, :confirmation_token,   unique: true
+    # add_index :users, :unlock_token,       unique: true
```
Comment (mesmerze) [inline db/migrate/20190805091526_devise_token_auth_create_users.rb]: remove or uncomment

File: spec/requests/sign_in_out_spec.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  it 'should get error status after sign_out' do
+    delete '/auth/sign_out'
+    expect(response).to have_http_status(404)
+
+    delete '/auth/sign_out', params: {
+      'access-token': 'vMzlOcJkftWkaU382TO2bg',
+      'token-type': 'Bearer',
+      'client': 'OWd3ZdcglX5nXDjg5bYEbA',
+      'expiry': '1566226122',
+      'uid': 'example@mail.ru'
```
Comment (mesmerze) [inline spec/requests/sign_in_out_spec.rb]: this is valid or invalid params? why 404? Is this expected behavior?

File: spec/requests/sign_up_spec.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+  it 'should get error status' do
+    post '/auth', params: { email: '', password: 'password' }
+    expect(response).to have_http_status(422)
+
+    post '/auth', params: { email: 'example@gmail.com', password: '12345' }
+    expect(response).to have_http_status(422)
+
+    post '/auth', params: { email: 'example@gmail.', password: '123456' }
+    expect(response).to have_http_status(422)
```
Comment (mesmerze) [inline spec/requests/sign_up_spec.rb]: next time use shared_examples for such similar specs

File: spec/requests/sign_in_out_spec.rb (hunk 6 above)
Comment (MargaretOm) [reply spec/requests/sign_in_out_spec.rb]: of course invalid

Comment (mesmerze) [reply spec/requests/sign_in_out_spec.rb]: if so put them into rspec [let](https://relishapp.com/rspec/rspec-core/v/2-5/docs/helper-methods/let-and-let) and call it like `:invalid_params`

File: app/javascript/components/SignIn.jsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+      })
+  };
+
+  render(){
+    return (
+      <div className="jumbotron">
+        <form onSubmit={this.handleSignIn}>
+          <div>
+            <label htmlFor='email'>E-mail</label><br />
+            <input type='text' name='email' ref={(input) => this.email = input} />
```
Comment (LuginaJulia) [inline app/javascript/components/SignIn.jsx]: Maybe you can use onChange?

File: db/migrate/20190805091526_devise_token_auth_create_users.rb [hunk 9]
```diff
@@ -0,0 +1,52 @@
+# :reek:FeatureEnvy
```
Comment (LuginaJulia) [inline db/migrate/20190805091526_devise_token_auth_create_users.rb]: Maybe it's unnecessary

File: db/seeds.rb [hunk 10]
```diff
@@ -5,3 +5,8 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+users = [
+    { email: 'user@gmail.com', password: '123456' }
+]
+
+User.create(users)
```
Comment (LuginaJulia) [inline db/seeds.rb]: EOF

File: app/javascript/components/App.jsx [hunk 11]
```diff
@@ -1,15 +1,36 @@
...
+  handleSignOut = (e) =>{
+    e.preventDefault();
+    Axios({
+      method: 'delete',
+      url: '/auth/sign_out',
+      data: JSON.parse(sessionStorage.user)
+    }).then(function (response) {
+        window.location = "/"
+      }).catch(function(error) {
+        window.location = "/#/sign_out"
```
Comment (ermaac) [inline app/javascript/components/App.jsx]: I'm not sure that is a good idea. you are trying to perform 'GET' request which was originally 'DELETE'. there is no such route which matches delete /sign_out

File: db/seeds.rb [hunk 12]
```diff
@@ -5,3 +5,8 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+users = [
+    { email: 'user@gmail.com', password: '123456' }
```
Comment (anatoliliotych) [inline db/seeds.rb]: looks not so full as it could be, try to create really good seeds, which could be used by others.

File: Gemfile [hunk 13]
```diff
@@ -45,6 +45,7 @@ gem 'bootsnap', '>= 1.1.0', require: false
...
+  gem 'factory_bot_rails'
```
Comment (ermaac) [inline Gemfile]: what about versions?

File: spec/requests/sign_in_out_spec.rb [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+  it 'should get success status after sign_in' do
+    get '/#/sign_in'
+    expect(response).to have_http_status(:success)
+
+    post '/auth/sign_in', params: { email: @user.email, password: @user.password }
+    expect(response).to have_http_status(:success)
+  end
+
+  it 'should get error status after sign_in' do
```
Comment (anatoliliotych) [inline spec/requests/sign_in_out_spec.rb]: strange test, could you please explain meaning? and what do you want to test?

File: spec/requests/sign_in_out_spec.rb [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+
+  it 'should get success status after sign_out' do
+    get '/'
+    expect(response).to have_http_status(:success)
+
+    delete '/auth/sign_out', params: @user.create_new_auth_token
+    expect(response).to have_http_status(:success)
+  end
+
+  it 'should get error status after sign_out' do
```
Comment (anatoliliotych) [inline spec/requests/sign_in_out_spec.rb]: it needs explanation

File: spec/requests/sign_up_spec.rb [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+  end
+  it 'should get success status' do
+    get '/#/sign_up'
+    expect(response).to have_http_status(:success)
+
+    post '/auth', params: { email: @user.email, password: @user.password }
+    expect(response).to have_http_status(:success)
+  end
+
+  it 'should get error status' do
```
Comment (anatoliliotych) [inline spec/requests/sign_up_spec.rb]: strange test, what do you test here?

File: app/javascript/components/App.jsx [hunk 17]
```diff
@@ -1,15 +1,36 @@
...
 
 class App extends Component {
+  handleSignOut = (e) =>{
+    e.preventDefault();
+    Axios({
+      method: 'delete',
+      url: '/auth/sign_out',
+      data: JSON.parse(sessionStorage.user)
+    }).then(function (response) {
+        window.location = "/"
```
Comment (ermaac) [inline app/javascript/components/App.jsx]: this is not 'React way' of redirecting. please investigate

File: app/javascript/components/SignUp.jsx [hunk 18]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import Axios from 'axios';
+
+export default class SignUp extends React.Component{
+  handleSignUp = (e) =>{
```
Comment (ermaac) [inline app/javascript/components/SignUp.jsx]: missing space)

File: app/javascript/components/SignUp.jsx [hunk 19]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import Axios from 'axios';
+
+export default class SignUp extends React.Component{
```
Comment (ermaac) [inline app/javascript/components/SignUp.jsx]: space

File: app/javascript/components/SignIn.jsx [hunk 20]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import Axios from 'axios';
+
+export default class SignIn extends React.Component{
```
Comment (ermaac) [inline app/javascript/components/SignIn.jsx]: space

File: db/migrate/20190805091526_devise_token_auth_create_users.rb [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+      t.datetime :confirmation_sent_at
+      t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
+      # t.integer  :failed_attempts, :default => 0, :null => false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+      ## User Info
+      t.string :name
```
Comment (ermaac) [inline db/migrate/20190805091526_devise_token_auth_create_users.rb]: there is some mismatch with our db design in documentation, please update the migration

File: db/seeds.rb (hunk 12 above)
Comment (ermaac) [inline db/seeds.rb]: indentation?

File: spec/requests/sign_in_out_spec.rb [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+    delete '/auth/sign_out'
+    expect(response).to have_http_status(404)
+
+    delete '/auth/sign_out', params: {
+      'access-token': '',
+      'token-type': '',
+      'client': '',
+      'expiry': '',
+      'uid': ''
+    }
```
Comment (mesmerze) [inline spec/requests/sign_in_out_spec.rb]: stop amending my comments!

Comment (mesmerze) [reply spec/requests/sign_in_out_spec.rb]: https://github.com/rubizza-camp/nicher-events/pull/5#discussion_r311885682

Comment (mesmerze) [reply spec/requests/sign_in_out_spec.rb]: https://github.com/rubizza-camp/nicher-events/pull/5#discussion_r311885682

File: db/migrate/20190805091526_devise_token_auth_create_users.rb (hunk 5 above)
Comment (mesmerze) [reply db/migrate/20190805091526_devise_token_auth_create_users.rb]: why it is resolved?

File: app/javascript/components/App.jsx (hunk 11 above)
Comment (MargaretOm) [reply app/javascript/components/App.jsx]: it's typo, in this line need write url "/"

File: app/javascript/components/App.jsx (hunk 17 above)
Comment (MargaretOm) [reply app/javascript/components/App.jsx]: need to do redirecting in the controller? but we use devise-token-auth and now we do not have a controller, it is built-in.

So need create controller, do override. And this all for only rediracting. I think it's bad way

File: db/seeds.rb [hunk 23]
```diff
@@ -5,3 +5,24 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
```
Comment (anatoliliotych) [inline db/seeds.rb]: remove these comments

File: app/javascript/components/SignIn.jsx [hunk 24]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import Axios from 'axios';
+
+export default class SignIn extends React.Component {
```
Comment (ermaac) [inline app/javascript/components/SignIn.jsx]: same rule as in ruby: class name should be a noun (e.g SigInForm)

Comment (ermaac) [inline app/javascript/components/SignIn.jsx]: Please check all components

File: db/migrate/20190805091526_devise_token_auth_create_users.rb [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+      # t.integer  :failed_attempts, :default => 0, :null => false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+      ## User Info
+      t.string :first_name
+      t.string :last_name
+      t.string :phone
+      t.string :nickname
+      t.string :image
```
Comment (ermaac) [inline db/migrate/20190805091526_devise_token_auth_create_users.rb]: 1. there is no such field in our db design
2. why image is a string? that is confusing. did you mean image_url?

File: db/migrate/20190805091526_devise_token_auth_create_users.rb [hunk 26]
```diff
@@ -0,0 +1,53 @@
...
+      t.json :tokens
+
+      t.timestamps
+    end
+
+    add_index :users, :email,                unique: true
+    add_index :users, [:uid, :provider],     unique: true
+    add_index :users, :reset_password_token, unique: true
+    add_index :users, :confirmation_token,   unique: true
+    # add_index :users, :unlock_token,       unique: true
```
Comment (ermaac) [inline db/migrate/20190805091526_devise_token_auth_create_users.rb]: Comment is not solved. Please do something with this (seems it needs to be removed)

File: app/javascript/components/SignIn.jsx [hunk 27]
```diff
@@ -0,0 +1,56 @@
...
+      JSON.stringify({
+        'access-token': response.request.getResponseHeader('access-token'),
+        'token-type': response.request.getResponseHeader('token-type'),
+        client: response.headers["client"],
+        expiry: response.headers["expiry"],
+        uid: response.headers["uid"]
+      }));
+        this.props.history.push('/');
+      }).catch(error => {
+        this.props.history.push(`/#/sign_in`);
```
Comment (ermaac) [inline app/javascript/components/SignIn.jsx]: Can you tell me what adds this sign '#'? Can we avoid this? Looks no so pretty

File: app/javascript/components/SignIn.jsx [hunk 28]
```diff
@@ -0,0 +1,56 @@
...
+    this.setState({ [user.target.name]: user.target.value });
+  }
+
+  render() {
+    return (
+      <div>
+        <form onSubmit={this.handleSignIn}>
+          <div>
+            <label htmlFor='email'>E-mail</label><br />
+            <input type="text" name="email" value={this.state.email} onChange={this.handleChange} className="form-control" />
```
Comment (LuginaJulia) [inline app/javascript/components/SignIn.jsx]: Maybe better put user's field under event key ```this.state.user.email ```

File: config/routes.rb [hunk 29]
```diff
@@ -2,5 +2,7 @@
   root 'welcome#index'
   get '*page', to: 'static#index',
                constraints: ->(req) { !req.xhr? && req.format.html? }
+  root 'welcome#index'
```
Comment (M1ha11) [inline config/routes.rb]: delete it

File: app/javascript/components/App.jsx [hunk 30]
```diff
@@ -1,15 +1,40 @@
...
+    super(props);
+  }
+
+  handleSignOut = (e) => {
+    e.preventDefault();
+    Axios({
+      method: 'delete',
+      url: '/auth/sign_out',
+      data: JSON.parse(sessionStorage.user)
+    }).then(function (response) {
```
Comment (LuginaJulia) [inline app/javascript/components/App.jsx]: ```(response) => { this.props.history.push('/'); }```  ?

File: spec/requests/sign_up_spec.rb [hunk 31]
```diff
@@ -0,0 +1,54 @@
...
+describe 'SignUp', type: :request do
+  let!(:user) { build(:user) }
+
+  context 'should get success status when sign_up' do
+    it 'get sign_up page' do
+      get '/#/sign_up'
+      expect(response).to have_http_status(:success)
+    end
+    it 'with right params' do
+      post '/auth', params: { first_name: user.first_name, last_name: user.last_name, email: user.email,
```
Comment (LuginaJulia) [inline spec/requests/sign_up_spec.rb]: ``` params: { user: user.attributes } ```

File: app/javascript/components/SignIn.jsx [hunk 32]
```diff
@@ -0,0 +1,58 @@
...
+    this.handleSignIn = this.handleSignIn.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleSignIn = (e) => {
+    e.preventDefault();
+    Axios({
+      method: 'post',
+      url: '/auth/sign_in',
+      data: this.state
```
Comment (MargaretOm) [inline app/javascript/components/SignIn.jsx]: this.state.user

File: app/javascript/packs/history.jsx [hunk 33]
```diff
@@ -0,0 +1,4 @@
+import { createBrowserHistory } from 'history';
+
+const customHistory = createBrowserHistory()
+export default customHistory;
```
Comment (MargaretOm) [inline app/javascript/packs/history.jsx]: newline

File: app/javascript/components/App.jsx [hunk 34]
```diff
@@ -22,15 +50,17 @@ const Navigation = () => (
...
+      <li className="nav-item"><NavLink exact className="nav-link" activeClassName="active" to="/sign_up">Sign up</NavLink></li>
+      <li className="nav-item"><NavLink exact className="nav-link" activeClassName="active" to="/sign_in">Sign in</NavLink></li>
     </ul>
   </nav>
 );
 
 const Main = () => (
   <Switch>
-    <Route exact path="/" />
+    <Route exact path="/"  />
```
Comment (ermaac) [inline app/javascript/components/App.jsx]: extra space)

File: app/javascript/components/App.jsx [hunk 35]
```diff
@@ -1,15 +1,43 @@
 import React, { Component } from 'react';
 import Home from './Home';
+import SignUp from './SignUp';
+import SignIn from './SignIn';
+import Axios from 'axios';
 import {HashRouter as Router, Route, NavLink, Switch} from 'react-router-dom'
+import history from "../packs/history";
+
```
Comment (ermaac) [inline app/javascript/components/App.jsx]: please leave 1 blank line

File: app/javascript/components/SignIn.jsx [hunk 36]
```diff
@@ -0,0 +1,58 @@
...
+    }).then(response => {
+      sessionStorage.setItem('user',
+      JSON.stringify({
+        'access-token': response.request.getResponseHeader('access-token'),
+        'token-type': response.request.getResponseHeader('token-type'),
+        client: response.headers["client"],
+        expiry: response.headers["expiry"],
+        uid: response.headers["uid"]
+      }));
+        this.props.history.push('/');
```
Comment (ermaac) [inline app/javascript/components/SignIn.jsx]: indentation issue

File: app/javascript/components/SignIn.jsx [hunk 37]
```diff
@@ -0,0 +1,58 @@
...
+      }).catch(error => {
+        this.props.history.push('/sign_in');
+      })
+  };
+
+  handleChange = (user) => {
+    const current_user = this.state.user;
```
Comment (ermaac) [inline app/javascript/components/SignIn.jsx]: camel case

File: app/javascript/components/SignIn.jsx [hunk 38]
```diff
@@ -0,0 +1,58 @@
...
+      <div>
+        <form onSubmit={this.handleSignIn}>
+          <div>
+            <label htmlFor='email'>E-mail</label><br />
+            <input type="text" name="email" value={this.state.user.email} onChange={this.handleChange} className="form-control" />
+          </div>
+
+          <div>
+            <label htmlFor='password'>Password</label><br />
+            <input type="text" name="password" value={this.state.user.password} onChange={this.handleChange} className="form-control" />
```
Comment (ermaac) [inline app/javascript/components/SignIn.jsx]: please change type to 'password'

File: app/javascript/components/SignUp.jsx [hunk 39]
```diff
@@ -0,0 +1,73 @@
...
+        uid: response.headers["uid"]
+      }));
+        this.props.history.push('/');
+      }).catch(error => {
+        this.props.history.push('/sign_up');
+      })
+  };
+
+  handleChange = (user) => {
+    const current_user = this.state.user;
```
Comment (ermaac) [inline app/javascript/components/SignUp.jsx]: camel case

File: app/javascript/packs/application.jsx [hunk 40]
```diff
@@ -20,5 +20,13 @@ console.log('Hello World from Webpacker')
 import React from 'react';
 import ReactDOM from 'react-dom';
 import App from '../components/App';
+import history from './history';
+import { Router, Route } from 'react-router-dom';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+// const history = createBrowserHistory();
```
Comment (ermaac) [inline app/javascript/packs/application.jsx]: useless comment

File: spec/requests/sign_in_out_spec.rb [hunk 41]
```diff
@@ -0,0 +1,58 @@
...
+  include ActionController::RespondWith
+  let(:user) { create(:user) }
+
+  context 'should get success status when sign_in' do
+    it 'get sign_in form' do
+      get '/#/sign_in'
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'with right params' do
```
Comment (ermaac) [inline spec/requests/sign_in_out_spec.rb]: it is not clear from the name what does this test check
please add appropriate verb to the test's name
it '<verb>s ...'

File: spec/requests/sign_in_out_spec.rb [hunk 42]
```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+describe 'SignInOut', type: :request do
+  include ActionController::RespondWith
+  let(:user) { create(:user) }
+
+  context 'should get success status when sign_in' do
+    it 'get sign_in form' do
+      get '/#/sign_in'
```
Comment (ermaac) [inline spec/requests/sign_in_out_spec.rb]: can you explain what adds this sign '#'. Can we avoid it?

File: app/javascript/components/App.jsx [hunk 43]
```diff
@@ -1,22 +1,58 @@
...
 
+const RegisterNavigation = () => (
+    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
+        <ul className="navbar-nav mr-auto">
+            <li className="nav-item"><NavLink exact className="nav-link" activeClassName="active" to="/sign_up">Sign up</NavLink></li>
+            <li className="nav-item"><NavLink exact className="nav-link" activeClassName="active" to="/sign_in">Sign in</NavLink></li>
+        </ul>
+    </nav>
+);
+const SignOUtNavigation = () => (
```
Comment (ermaac) [inline app/javascript/components/App.jsx]: please make u in Out non-capital

File: app/javascript/components/App.jsx [hunk 44]
```diff
@@ -1,22 +1,58 @@
...
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state ={};
+  }
 
-class App extends Component {
   render() {
+    let registerNavigation;
```
Comment (ermaac) [inline app/javascript/components/App.jsx]: I don't think it is a good naming for variable. It takes as value either <RegisterNavigation /> (same as variable) or <SignOUtNavigation />. To my opinion navbarComponent looks better

File: app/javascript/components/SignIn.jsx [hunk 45]
```diff
@@ -0,0 +1,71 @@
...
+      sessionStorage.setItem('user',
+      JSON.stringify({
+        'access-token': response.request.getResponseHeader('access-token'),
+        'token-type': response.request.getResponseHeader('token-type'),
+        client: response.headers["client"],
+        expiry: response.headers["expiry"],
+        uid: response.headers["uid"]
+      }));
+      this.props.history.push('/');
+      }).catch(error => {
```
Comment (ermaac) [inline app/javascript/components/SignIn.jsx]: indentation issue

File: app/javascript/components/SignIn.jsx [hunk 46]
```diff
@@ -0,0 +1,71 @@
...
+        {this.state.errors.map((error) => {
+          return(
+              <p>{error}</p>
+          )
+        })}
+      </div>
+    }
+    return (
+      <div>
+        <div>
```
Comment (ermaac) [inline app/javascript/components/SignIn.jsx]: For better readability I'd add className like 'errors'
Also it is worth choosing better name for variable (e.g. errorMessages)

File: app/javascript/components/SignOut.jsx [hunk 47]
```diff
@@ -0,0 +1,32 @@
...
+import Axios from "axios";
+
+export default class SignOut extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state ={ hasError: false };
+    }
+
+    componentDidMount() {
+            Axios({
```
Comment (ermaac) [inline app/javascript/components/SignOut.jsx]: please fix indentation

File: app/javascript/components/SignUp.jsx [hunk 48]
```diff
@@ -0,0 +1,87 @@
...
+    }).then(response => {
+      sessionStorage.setItem('user',
+      JSON.stringify({
+        'access-token': response.request.getResponseHeader('access-token'),
+        'token-type': response.request.getResponseHeader('token-type'),
+        client: response.headers["client"],
+        expiry: response.headers["expiry"],
+        uid: response.headers["uid"]
+      }));
+        this.props.history.push('/');
```
Comment (ermaac) [inline app/javascript/components/SignUp.jsx]: indentation

File: app/javascript/components/App.jsx [hunk 49]
```diff
@@ -1,22 +1,57 @@
...
+  constructor(props) {
+    super(props);
+    this.state ={};
+  }
+
   render() {
+    let navbarComponent;
+    if (sessionStorage.user == null){
+        navbarComponent = <RegisterNavigation />
+    }else
```
Comment (ermaac) [inline app/javascript/components/App.jsx]: space before else

File: app/javascript/components/App.jsx [hunk 50]
```diff
@@ -1,22 +1,57 @@
...
+    {
```
Comment (ermaac) [inline app/javascript/components/App.jsx]: please put this bracket on previous line

File: app/javascript/components/SignOut.jsx [hunk 51]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import Axios from "axios";
+
+export default class SignOut extends React.Component {
+    constructor(props) {
```
Comment (ermaac) [inline app/javascript/components/SignOut.jsx]: indentation issue

File: app/javascript/components/SignOut.jsx [hunk 52]
```diff
@@ -0,0 +1,32 @@
...
+      method: 'delete',
+      url: '/auth/sign_out',
+      data: JSON.parse(sessionStorage.user)
+    }).then(response => {
+        sessionStorage.removeItem('user');
+        this.props.history.push('/');
+      }).catch(error => {
+          sessionStorage.removeItem('user');
+          this.setState({ hasError: true })
+        })
```
Comment (ermaac) [inline app/javascript/components/SignOut.jsx]: indentation issue

**Pull request merged**